tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -2639,6 +2640,10 @@ TEST_F(CompilerTest, CodeGenSamples){
   CodeGenTestCheckBatchDir(L"Samples");
 }
 
+TEST_F(CompilerTest, AllocaAfterStaticInit) {
+  CodeGenTest(L"gv_memcpy_before_alloca.hlsl");
```

Comment:
We shouldn't be using `CodeGenTest`, it doesn't even run FileCheck, only asserts that the disassembly is non-empty. Also let's leverage batch tests. This should go into `batch/passes/sroa_hlsl`